tracer/src/Datadog.Trace/Configuration/GlobalSettings.cs

```diff
@@ -86,18 +91,18 @@ public static void SetDebugEnabled(bool enabled)
         public static void Reload()
         {
             DatadogLogging.Reset();
-            Source = FromDefaultSources();
+            Instance = FromDefaultSources();
+            ConfigurationSource = CreateDefaultConfigurationSource();
```

Comment:
We should swap these - `FromDefaultSources()` uses `ConfigurationSource`, so it wouldn't actually reload the values:
```suggestion
            ConfigurationSource = CreateDefaultConfigurationSource();
            Instance = FromDefaultSources();
```

Reply:
I've extracted `ConfigurationSource` from `GlobalSettings` and changed the order here 45afa07cdc2e0dc1b5c76a1e65b93fd9c34fc07e